### PR TITLE
ref(docs): update to deploy/run via oci

### PR DESCRIPTION
**Pull request by vdice**
* Updates the deployment mechanism of this website (GH workflow and nomad job) to use OCI

TODO:
- [x] Test run from this branch

### CODE REVIEW

**Comment by vdice** [comment]
This is ready for review.  Website is successfully deployed from this branch (eg https://github.com/fermyon/bartholomew/actions/runs/7023277511) and I toggled DNS to the LB hosting this oci-based deployment.  (I can toggle back if there happen to be other PRs in this repo affecting the docs site, but I'm expecting there won't be any.)

Note that this PR represents the pattern that'll be used to convert the other Fermyon websites, so definitely want to get things right/address nits/etc.

After LGTMs and prior to merge, I'll squash/rebase commits and remove my testing branch from the workflow trigger list.

File: deploy/bartholomew-docs.nomad [hunk 1]
```diff
@@ -98,31 +70,58 @@ job "bartholomew-docs" {
...
+          --listen ${NOMAD_IP_http}:${NOMAD_PORT_http} \
+          --log-dir ${NOMAD_ALLOC_DIR}/logs \
+          --temp ${NOMAD_ALLOC_DIR}/tmp \
+          -e BASE_URL=${BASE_URL}
+        EOF
+      }
+
+      template {
+        destination = "${NOMAD_SECRETS_DIR}/env.txt"
+        change_mode = "restart"
```
Comment (vdice) [inline deploy/bartholomew-docs.nomad]: Actually, we should probably change this to "noop", right?  Once we've pulled the image and run `spin up`, we don't need to restart due to expired creds.

I'd either want to go "noop" or increase the ttl, more pertinently for heavier-traffic sites like developer and fermyon.com...

Comment (vdice) [reply deploy/bartholomew-docs.nomad]: (Updated to use `noop` in https://github.com/fermyon/bartholomew/pull/185/commits/bd941125ec00ac7ace9667835b282f5113a77775)

**Review by adamreese — APPROVED** [review]

**Comment by vdice** [comment]
(As mentioned, https://github.com/fermyon/bartholomew/pull/185/commits/93e7e82200a723456949fc6a259a053a2eb1341e represents commits squashed down to one and removal of the `vdice*` testing branch config from deploy.yml)

**Pull request merged**